include/diaspora.php

```diff
@@ -2031,7 +2092,7 @@ private function original_item($guid, $orig_author, $author) {
 			if (self::is_reshare($r[0]["body"], true))
 				$r = array();
 			elseif (self::is_reshare($r[0]["body"], false)) {
-				$r[0]["body"] = diaspora2bb(bb2diaspora($r[0]["body"]));
+				$r[0]["body"] = self::replace_people_guid(diaspora2bb(bb2diaspora($r[0]["body"])), $r[0]["author-link"]);
```

Comment:
`self::replace_people_guid()` just is a `preg_replace_callback` performing a string replacement, so we don't need to check for valid return values: it returns a string whether it performed one or more replacements or not.

I would love to use Exceptions, however I don't think it is planned in the near future for Friendica.

Agreed on chaining, I can break down the calls.